scripts/karma.coffee

```diff
@@ -48,20 +47,59 @@ class Karma
       delete @cache.things[thing]
     @robot.brain.data.karma = @cache
 
+  cleanKarmaHistory: (actor) ->
+    if @khistory[actor.id]?
+      DAY = 1000 * 60 * 60  * 24
+      today = new Date()
+      @khistory[actor.id] = (x for x in @khistory[actor.id] when (Math.round((today.getTime() - x) / DAY)) <= 7 )
+
+  getKarmaHistory: (actor) ->
+    if @khistory[actor.id]?
+      DAY = 1000 * 60 * 60  * 24
+      today = new Date()
+      todayKarma = (x for x in @khistory[actor.id] when (Math.round((today.getTime() - x) / DAY)) == 0 )
+      yesterdayKarma = (x for x in @khistory[actor.id] when (Math.round((today.getTime() - x) / DAY)) == 1 )
```

Comment:
Nitpicking, but are these today and yesterday or the last 24 hours and the previous 24 hours before that? Asking more so that it can be explained in the event that someone asks about the process.